fetch/src/test/java/com/indix/gocd/s3fetch/FetchExecutorTest.java

```diff
@@ -87,14 +112,148 @@ public void shouldBeSuccessResultOnSuccessfulFetch() {
         assertTrue(executionResult.isSuccessful());
         assertThat(executionResult.getMessagesForDisplay(), is("Fetched all artifacts"));
         verify(mockStore, times(1)).getPrefix("TestPublish/defaultStage/defaultJob/20.1", "./artifacts");
+    }
+
+
+
+    @Test
+    public void shouldUnzipArchiveIfFetchReturnsOneFileThatIsArtifactsZip() {
```

Comment:
Definitely think, the tests could be refactored heavily. Will spend more time and send notes on this.